mLRS/Common/time_stats.h

```diff
@@ -0,0 +1,83 @@
+//*******************************************************
+// Copyright (c) MLRS project
+// GPL3
+// https://www.gnu.org/licenses/gpl-3.0.de.html
+// OlliW @ www.olliw.eu
+//*******************************************************
+// time_stats
+//*******************************************************
+#ifndef TIME_STATS_H
+#define TIME_STATS_H
+#pragma once
+
+// Time Statistics for up to PM_NUM_BLOCK pairs of measurement points
+#ifdef USE_DEBUG
+#define TS_NUM_BLOCK 10
+
+struct TS_store {
+    uint32_t start;
+    uint32_t last_report;
+    uint32_t total;
+    uint32_t min;
+    uint32_t max;
+    uint32_t count;
+    uint32_t overflow;
+    uint16_t last_cnt;
+};
+static struct TS_store TS_data[TS_NUM_BLOCK];
+
+inline uint32_t TS_MICROS32(int block) {
+
+    // Separate overflow and last_cnt for each block allows either ISR/normal use, but not both
```

Comment:
You didn't comment on this in a timer.h, but, in fact, while micros16() is safe to call from an ISR, micros64() is not safe to call from an ISR.  If we want the ability to use TS functions in both user level and ISR level code, we need to avoid race conditions. If we have a single copy of overflow_cnt and last_cnt like you do in micros64(), it won't always count properly when called from both normal and ISR level because an interrupt can happen in the middle of the call to micros64() and a second call to micros64() can happen which might result in an incorrect overflow_cnt or last_cnt.  It's faster to have a separate set of variables for each code block than to disable interrupts in the critical section.  If you are wondering if the race actually shows up in the statistics, I can testify that it did when I tried to use my first version of this code to gather data on interrupt timing.  I was seeing output every 3 seconds instead of every 10 seconds.

Reply:
Note that with my approach, it is still not safe to split TS_START() and TS_END() between normal and ISR code for the same block.  Thus, we can't use these functions to measure the time between setting a device register at user level and the interrupt which results from this, for example.  Any given block must be referenced only in a single ISR or only in user level code.

Reply:
thx for the explanation. 
I guess it would be useful to have a bit of comments